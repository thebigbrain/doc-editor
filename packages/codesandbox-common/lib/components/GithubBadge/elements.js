"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledA = exports.Icon = exports.Text = exports.BorderRadius = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  text-decoration: none;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  padding: 3px 5px;\n  background-color: #4f5459;\n  border-radius: 2px;\n  color: ", ";\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n\n  color: ", ";\n  border-radius: 4px;\n  padding: 3px 5px;\n\n  &:hover {\n    color: rgba(255, 255, 255, 0.6);\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      &:hover {\n        background-color: #4f5459;\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  transition: 0.3s ease all;\n  border-radius: 4px;\n  border: 1px solid #4f5459;\n  font-size: 0.75em;\n  margin-right: 1rem;\n\n  display: flex;\n\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var BorderRadius = _styledComponents["default"].div(_templateObject(), function (props) {
  return props.hasUrl && (0, _styledComponents.css)(_templateObject2());
});

exports.BorderRadius = BorderRadius;

var Text = _styledComponents["default"].span(_templateObject3(), function (props) {
  return props.theme.light ? '#636363' : 'rgba(255, 255, 255, 0.6)';
});

exports.Text = Text;

var Icon = _styledComponents["default"].span(_templateObject4(), function (props) {
  return props.theme.background;
});

exports.Icon = Icon;

var StyledA = _styledComponents["default"].a(_templateObject5());

exports.StyledA = StyledA;