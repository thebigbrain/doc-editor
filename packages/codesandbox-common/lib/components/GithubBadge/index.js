"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _markGithub = _interopRequireDefault(require("react-icons/lib/go/mark-github"));

var _elements = require("./elements");

var DivOrA = function DivOrA(_ref) {
  var href = _ref.href,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["href"]);
  return href ? _react["default"].createElement(_elements.StyledA, (0, _extends2["default"])({
    target: "_blank",
    rel: "noopener noreferrer",
    href: href
  }, props)) : _react["default"].createElement("div", props);
};

var GithubBadge = function GithubBadge(_ref2) {
  var username = _ref2.username,
      repo = _ref2.repo,
      url = _ref2.url,
      branch = _ref2.branch,
      props = (0, _objectWithoutProperties2["default"])(_ref2, ["username", "repo", "url", "branch"]);
  return _react["default"].createElement(DivOrA, (0, _extends2["default"])({}, props, {
    href: url
  }), _react["default"].createElement(_elements.BorderRadius, {
    hasUrl: Boolean(url)
  }, _react["default"].createElement(_elements.Icon, null, _react["default"].createElement(_markGithub["default"], null)), _react["default"].createElement(_elements.Text, null, username, "/", repo, branch && branch !== 'master' ? "@".concat(branch) : '')));
};

var _default = GithubBadge;
exports["default"] = _default;