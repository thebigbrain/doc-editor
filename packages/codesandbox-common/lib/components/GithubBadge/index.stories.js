"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@storybook/react");

var _ = _interopRequireDefault(require("."));

var stories = (0, _react2.storiesOf)('components/GithubBadge', module);
stories.add('Master', function () {
  return _react["default"].createElement(_["default"], {
    username: "CompuIves",
    repo: "codesandbox-client",
    branch: "master"
  });
}).add('Other Branch', function () {
  return _react["default"].createElement(_["default"], {
    username: "CompuIves",
    repo: "codesandbox-client",
    branch: "storybook"
  });
});