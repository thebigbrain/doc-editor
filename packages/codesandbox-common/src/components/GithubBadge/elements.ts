import styled, { css } from 'styled-components';

export const BorderRadius = styled.div<{ hasUrl?: boolean }>`
  transition: 0.3s ease all;
  border-radius: 4px;
  border: 1px solid #4f5459;
  font-size: 0.75em;
  margin-right: 1rem;

  display: flex;

  ${props =>
    props.hasUrl &&
    css`
      &:hover {
        background-color: #4f5459;
      }
    `};
`;

export const Text = styled.span`
  display: inline-block;

  color: ${props =>
    props.theme.light ? '#636363' : 'rgba(255, 255, 255, 0.6)'};
  border-radius: 4px;
  padding: 3px 5px;

  &:hover {
    color: rgba(255, 255, 255, 0.6);
  }
`;

export const Icon = styled.span`
  display: inline-block;
  padding: 3px 5px;
  background-color: #4f5459;
  border-radius: 2px;
  color: ${props => props.theme.background};
`;

export const StyledA = styled.a`
  text-decoration: none;
`;
