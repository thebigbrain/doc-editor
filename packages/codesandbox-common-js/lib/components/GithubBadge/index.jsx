import React from 'react';
import GithubIcon from 'react-icons/lib/go/mark-github';
import { BorderRadius, Text, Icon, StyledA } from './elements';
const DivOrA = ({ href, ...props }) => href ? (<StyledA target="_blank" rel="noopener noreferrer" href={href} {...props}/>) : (<div {...props}/>);
const GithubBadge = ({ username, repo, url, branch, ...props }) => (<DivOrA {...props} href={url}>
    <BorderRadius hasUrl={Boolean(url)}>
      <Icon>
        <GithubIcon />
      </Icon>
      <Text>
        {username}/{repo}
        {branch && branch !== 'master' ? `@${branch}` : ''}
      </Text>
    </BorderRadius>
  </DivOrA>);
export default GithubBadge;
