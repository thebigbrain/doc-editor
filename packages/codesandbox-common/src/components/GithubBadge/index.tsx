import React, { FunctionComponent } from 'react';
import GithubIcon from 'react-icons/lib/go/mark-github';
import { BorderRadius, Text, Icon, StyledA } from './elements';

type BadgeProps = {
  username: string;
  repo: string;
  url?: string;
  branch: string;
};

type DivOrAProps = {
  href?: string;
  children: React.ReactNode;
};

const DivOrA: FunctionComponent<DivOrAProps> = ({ href, ...props }) =>
  href ? (
    <StyledA target="_blank" rel="noopener noreferrer" href={href} {...props} />
  ) : (
    <div {...props} />
  );

const GithubBadge: FunctionComponent<BadgeProps> = ({
  username,
  repo,
  url,
  branch,
  ...props
}) => (
  <DivOrA {...props} href={url}>
    <BorderRadius hasUrl={Boolean(url)}>
      <Icon>
        <GithubIcon />
      </Icon>
      <Text>
        {username}/{repo}
        {branch && branch !== 'master' ? `@${branch}` : ''}
      </Text>
    </BorderRadius>
  </DivOrA>
);

export default GithubBadge;
