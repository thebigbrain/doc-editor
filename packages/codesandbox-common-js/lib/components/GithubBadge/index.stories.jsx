import React from 'react';
import { storiesOf } from '@storybook/react';
import GithubBadge from '.';
const stories = storiesOf('components/GithubBadge', module);
stories
    .add('Master', () => (<GithubBadge username="CompuIves" repo="codesandbox-client" branch="master"/>))
    .add('Other Branch', () => (<GithubBadge username="CompuIves" repo="codesandbox-client" branch="storybook"/>));
