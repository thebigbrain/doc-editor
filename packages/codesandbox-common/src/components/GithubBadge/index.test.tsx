import 'jest-styled-components';
import React from 'react';
import mountWithTheme from '../../test/themeMount';
import GithubBadge from '.';

describe('<GithubBadge /> rendering', () => {
  it('master', () => {
    const wrapper = mountWithTheme(
      <GithubBadge
        username="CompuIves"
        repo="codesandbox-client"
        branch="master"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('other branch', () => {
    const wrapper = mountWithTheme(
      <GithubBadge
        username="CompuIves"
        repo="codesandbox-client"
        branch="storybook"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
