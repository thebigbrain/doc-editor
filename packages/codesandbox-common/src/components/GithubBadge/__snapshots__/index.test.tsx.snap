// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GithubBadge /> rendering master 1`] = `ReactWrapper {}`;

exports[`<GithubBadge /> rendering other branch 1`] = `ReactWrapper {}`;
